### PR TITLE
Fix #2156: Do not tolerate bit 7 being on in US_ASCII decoder.

### LINKED ISSUE #2156: US_ASCII should not allow decoding bytes with bit 7 on
All JDK implementations I could try consistently report an error for such bytes. And I cannot find any source on US-ASCII supporting my original implementation, which was to ignore that bit.

### PR DESCRIPTION
**Pull request by sjrd**

### CODE REVIEW

**Comment by sjrd** [comment]
Review by @nicolasstucki

File: test-suite/shared/src/test/scala/org/scalajs/testsuite/niocharset/USASCIITest.scala [hunk 1]
```diff
@@ -26,14 +24,9 @@ class USASCIITest extends BaseCharsetTest(Charset.forName("US-ASCII")) {
...
 
-    if (!executingInJVM) {
-      // Bit 7 is ignored, giving the same results as above
-      testDecode(bb"c8 e5 ec ec ef")(cb"Hello")
-      testDecode(bb"c2 ef ee ea ef f5 f2")(cb"Bonjour")
-
-      testDecode(bb"80 81 8a 90 a0")(cb"\u0000\u0001\u000a\u0010 ")
-      testDecode(bb"ff ff")(cb"\u007f\u007f")
-    }
+    // Bit 7 on is an error
```
Comment (nicolasstucki) [inline test-suite/shared/src/test/scala/org/scalajs/testsuite/niocharset/USASCIITest.scala]: Mention that this is testing issue #2156.

File: test-suite/shared/src/test/scala/org/scalajs/testsuite/niocharset/USASCIITest.scala [hunk 2]
```diff
@@ -88,9 +81,8 @@ class USASCIITest extends BaseCharsetTest(Charset.forName("US-ASCII")) {
     assertTrue(encoder.isLegalReplacement(Array(0x00.toByte)))
     assertTrue(encoder.isLegalReplacement(Array(0x41.toByte)))
     assertTrue(encoder.isLegalReplacement(Array('?'.toByte)))
-    if (!executingInJVM) {
-      assertTrue(encoder.isLegalReplacement(Array(0x80.toByte)))
-      assertTrue(encoder.isLegalReplacement(Array(0xff.toByte)))
-    }
+
```
Comment (nicolasstucki) [inline test-suite/shared/src/test/scala/org/scalajs/testsuite/niocharset/USASCIITest.scala]: Same

**Comment by nicolasstucki** [comment]
Thats all

**Comment by sjrd** [comment]
Updated.

**Comment by nicolasstucki** [comment]
LGTM

**Comment by sjrd** [comment]
retest this please

**Pull request merged**